.github/workflows/apple.yml

```diff
@@ -40,135 +40,39 @@ jobs:
           name: pythonbuild
           path: target/release/pythonbuild
 
+  generate-matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.set-matrix.outputs.matrix }}
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: Set up Python
+        uses: astral-sh/setup-uv@v4
+
+      - name: Get pull request labels
+        id: get-labels
+        run: |
+          # Convert GitHub labels array to comma-separated string
+          LABELS=$(echo '${{ toJson(github.event.pull_request.labels.*.name) }}' | jq -r 'join(",")')
+          echo "labels=$LABELS" >> $GITHUB_OUTPUT
+
+      - name: Generate build matrix
+        id: set-matrix
+        run: |
+          uv run ci-matrix.py --platform darwin --labels "${{ steps.get-labels.outputs.labels }}" > matrix.json && echo "matrix=$(cat matrix.json)" >> $GITHUB_OUTPUT
```

Comment:
Do we need single quotes here to avoid shell injection/escape?

Reply:
As in, via the labels? I don't think people can create arbitrary labels in the organization, and I don't think commits on `main` could be targeted — but I'm happy to switch to single quotes to make it more robust anyway.